### PR TITLE
Various bug fixes, clean up, and features

### PR DESCRIPTION
**Pull request by cameronbowie**

### CODE REVIEW

File: NWDeafSchool/NWDeafSchool/Helpers/Shuffle.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+			while (n > 1)
+			{
+				n--;
+				int k = rng.Next(n + 1);
+				T value = list[k];
+				list[k] = list[n];
+				list[n] = value;
+			}
+
+            return list;
```
Comment (ddemas) [inline NWDeafSchool/NWDeafSchool/Helpers/Shuffle.cs]: Indentation

File: NWDeafSchool/NWDeafSchool/Models/WordCheckResult.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace NWDeafSchool
+{
+	public enum WordCheckResult
+	{
+		Correct = 1,
+		SpeechCorrect = 2,
+		SignCorrect = 3,
+        Incorrect = 4
```
Comment (ddemas) [inline NWDeafSchool/NWDeafSchool/Models/WordCheckResult.cs]: Indentation

File: NWDeafSchool/NWDeafSchool/Models/Word.cs [hunk 3]
```diff
@@ -12,14 +12,8 @@ public class Word
...
         public string Value { get; set; }
         public int SyllableCount { get; set; }
-        public int WordImageId { get; set; }
-        public int SignImageId { get; set; }
         public int Difficulty { get; set; }
-        //Deprecated
         public string PrimaryImageUrl { get; set; }
-        //Deprecated
         public string SeePrintUrl { get; set; }
-        //Deprecated
```
Comment (expwkuhn) [inline NWDeafSchool/NWDeafSchool/Models/Word.cs]: So these are not deprecate? I thought the API didn't return them

File: NWDeafSchool/NWDeafSchool/Views/MainPage.cs [hunk 4]
```diff
@@ -3,22 +3,42 @@
 
 using Plugin.Settings;
 using System.Windows.Input;
+using Plugin.Settings.Abstractions;
 
 namespace NWDeafSchool
 {
     public partial class MainPage : ContentPage
     {
+		private MainPageViewModel viewModel;
```
Comment (ddemas) [inline NWDeafSchool/NWDeafSchool/Views/MainPage.cs]: Indentation

File: NWDeafSchool/NWDeafSchool/Views/UnitsPage.xaml.cs [hunk 5]
```diff
@@ -34,12 +35,17 @@ public UnitsPage(Student student, SessionTypeValue type)
 
             IEnumerable<Word> words = await viewModel.LoadWords(unit.Id);
 
-            await Navigation.PushAsync(new WordPage(student, type, unit, words));
-
-            Debug.WriteLine("Selected unit: " + unit.Name);
+			if (words.Any())
```
Comment (ddemas) [inline NWDeafSchool/NWDeafSchool/Views/UnitsPage.xaml.cs]: Indentation

File: NWDeafSchool/NWDeafSchool/Views/WordPage.xaml.cs [hunk 6]
```diff
@@ -50,51 +50,98 @@ public WordPage(Student student, SessionTypeValue type, Unit unit, IEnumerable<W
...
 
-        void OnClickSkipForever(object sender, EventArgs args)
+        async void OnClickSkipForever(object sender, EventArgs args)
         {
-            Console.WriteLine("skip forever");
+            var answer = await DisplayAlert("Are you sure?", "If you skip, this word will not appear for this student ever again.", "Yes", "No");
+            if (answer) 
+            {
+
+				NextWord(ScoreValue.Skipped);
```
Comment (ddemas) [inline NWDeafSchool/NWDeafSchool/Views/WordPage.xaml.cs]: whitespace

File: NWDeafSchool/NWDeafSchool/Models/Word.cs (hunk 3 above)
Comment (ddemas) [reply NWDeafSchool/NWDeafSchool/Models/Word.cs]: We talked with the API folks and they added them to the return type

Comment (expwkuhn) [reply NWDeafSchool/NWDeafSchool/Models/Word.cs]: Sounds good.

**Pull request merged**